### PR TITLE
add  fence trtanslation for  zh_CN

### PR DESCRIPTION
**Pull request by luckhuanhuan**

### CODE REVIEW

**Comment by wenningerk** [comment]
test this please

File: daemons/fenced/pacemaker-fenced.c [hunk 1]
```diff
@@ -1377,15 +1377,216 @@ st_peer_update_callback(enum crm_status_type type, crm_node_t * node, const void
     }
 }
 
+static pcmk__cluster_option_t fenceder_options[] = {
```
Comment (kgaillot) [inline daemons/fenced/pacemaker-fenced.c:1380]: "fenced_options" or "fencer_options" would be a bit better

("fenced" comes from the daemon name pacemaker-fenced, while "fencer" describes its function)

File: daemons/fenced/pacemaker-fenced.c [hunk 2]
```diff
@@ -1377,15 +1377,216 @@ st_peer_update_callback(enum crm_status_type type, crm_node_t * node, const void
...
+    /* name, old name, type, allowed values,
+     * default value, validator,
+     * short description,
+     * long description
+     */
+    {
+        PCMK_STONITH_HOST_ARGUMENT, NULL, "string", NULL, "port", NULL,
+        N_("dvanced use only: An alternate parameter to supply instead of 'port'"),
```
Comment (kgaillot) [inline daemons/fenced/pacemaker-fenced.c:1388]: spelling: "Advanced"

File: daemons/fenced/pacemaker-fenced.c [hunk 3]
```diff
@@ -1377,15 +1377,216 @@ st_peer_update_callback(enum crm_status_type type, crm_node_t * node, const void
...
+	"pcmk_status_retries",NULL, "integer", NULL, "2", NULL,
+	N_("Advanced use only: The maximum number of times to retry the 'status' command within the timeout period"),
+        N_("Some devices do not support multiple connections."
+           " Operations may 'fail' if the device is busy with another task so Pacemaker will automatically retry the operation,      if there is time remaining."
+           " Use this option to alter the number of times Pacemaker retries 'status' actions before giving up.")
+    },
+};
+
+void
+crmd_metadata(void)
```
Comment (kgaillot) [inline daemons/fenced/pacemaker-fenced.c:1563]: "crmd" is the old name for the pacemaker-controld daemon; "fenced_metadata" or "fencer_metadata" would be better here

File: daemons/fenced/pacemaker-fenced.c [hunk 4]
```diff
@@ -1377,15 +1377,216 @@ st_peer_update_callback(enum crm_status_type type, crm_node_t * node, const void
...
+}
+
+/*
+static const char *
+fenceder_options(GHashTable *options, const char *name)
+{
+    return pcmk__cluster_option(options, fenceder_options,
+                                PCMK__NELEM(fenceder_options), name);
+}
+*/
```
Comment (kgaillot) [inline daemons/fenced/pacemaker-fenced.c:1582]: You can delete this commented out code

File: lib/common/options.c [hunk 5]
```diff
@@ -663,8 +664,16 @@ pcmk__format_option_metadata(const char *name, const char *desc_short,
...
+	if (flag)
+	{
+        	g_string_append_printf(s, "    <parameter name=\"%s\" unique=\"0\" >\n",
                                option_list[lpc].name);
+	}
+	else 
+	{
+        	g_string_append_printf(s, "    <parameter name=\"%s\">\n",
+                               option_list[lpc].name);
+	}
```
Comment (kgaillot) [inline lib/common/options.c:676]: Thanks for trying to keep the output the same, but the "unique" parameter was recently deprecated, so we don't need to show it here. You can revert this part and the new function argument.

**Review by kgaillot — COMMENTED** [review]
Hi,

Thanks for the thoughtful solution! This will help bring the fencer code up to date.

**Comment by kgaillot** [comment]
@gao-yan , would you have time to review the translations here?

**Comment by gao-yan** [comment]
> @gao-yan , would you have time to review the translations here?

Sure, will do.

**Comment by kgaillot** [comment]
> > @gao-yan , would you have time to review the translations here?
> 
> Sure, will do.

note that this got moved to #2865

**Pull request closed without merging**